nanshe_ipython.ipynb

```diff
@@ -328,6 +331,280 @@
+        "\n",
+        "    return find_best_match(matches)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "import dask\n",
+        "import dask.array\n",
+        "import dask.utils\n",
+        "\n",
+        "import dask.array as da\n",
+        "\n",
+        "import dask_ndfilters\n",
+        "import dask_ndfourier\n",
+        "\n",
+        "try:\n",
+        "    import pyfftw.interfaces.numpy_fft as numpy_fft\n",
+        "except ImportError:\n",
+        "    import numpy.fft as numpy_fft\n",
+        "\n",
+        "from dask.array.fft import fft_wrap\n",
+        "\n",
+        "fftn = fft_wrap(numpy_fft.fftn)\n",
+        "ifftn = fft_wrap(numpy_fft.ifftn)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "num_reps = 5\n",
+        "tmpl_hist_wght = 0.25\n",
+        "thld_rel_dist = 0.0\n",
+        "\n",
+        "norm_frames = 100\n",
+        "\n",
+        "\n",
+        "# Somehow we can't overwrite the file in the container so this is needed.\n",
+        "io_remove(data_basename + postfix_reg + zarr_ext)\n",
+        "io_remove(data_basename + postfix_reg + h5_ext)\n",
+        "\n",
+        "\n",
+        "with open_zarr(data_basename + zarr_ext, \"r\") as f:\n",
+        "    with get_executor(client) as executor:\n",
+        "        # Load and prep data for computation.\n",
+        "        imgs = f[\"images\"]\n",
+        "        da_imgs = da.from_array(imgs, chunks=imgs.chunks)\n",
+        "        da_imgs = da_imgs.rechunk((1,) + imgs.shape[1:])\n",
+        "\n",
+        "        da_imgs = da_imgs[1:-1]\n",
+        "\n",
+        "        da_imgs_flt = da_imgs\n",
+        "        if not (issubclass(da_imgs_flt.dtype.type, np.floating) and \n",
+        "                da_imgs_flt.dtype.itemsize >= 4):\n",
+        "            da_imgs_flt = da_imgs_flt.astype(np.float32)\n",
+        "\n",
+        "        # Create frame array\n",
+        "        frame_shape = da.from_array(\n",
+        "            np.array(da_imgs_flt.shape[1:], dtype=int),\n",
+        "            chunks=(da_imgs_flt.ndim - 1,)\n",
+        "        )\n",
+        "\n",
+        "        # Persist frame shape\n",
+        "        frame_shape = frame_shape.persist()\n",
+        "\n",
+        "        # Median filter frames\n",
+        "        da_imgs_flt = dask_ndfilters.median_filter(da_imgs_flt, 3)\n",
+        "\n",
+        "        # Compute the FFT of frames\n",
+        "        da_imgs_fft = fftn(da_imgs_flt, axes=tuple(irange(1, imgs.ndim)))\n",
+        "\n",
+        "        # High pass filter frames\n",
+        "        da_imgs_fft_smoothed = da.stack([\n",
+        "            dask_ndfourier.fourier_gaussian(da_imgs_fft[i], 100) for i in irange(len(da_imgs_fft))\n",
+        "        ])\n",
```

Comment:
Should change this to one filter on all dimensions by tacking on `0` for the first dimension's sigma. This should be more performant to compute than doing this stacking.